ApollosProject/apollos-apps
# feat: Hide flagged comments

## DESCRIPTION

### What does this PR do, or why is it needed?

Filters out comments with a number of flags which exceed the configured value. 

### What design trade-offs/decisions were made?

### How do I test this PR?

Add a comment: 
```
mutation addComment {
  addComment(text:"Flag this comment", parentId:"WeekendContentItem:f8460101368b38c5aa9bbea8b2b8a625") {
    id
  }.
}
```

Make sure it's there:
```
query content {
  node(id:"WeekendContentItem:f8460101368b38c5aa9bbea8b2b8a625") {
    ... on WeekendContentItem {
      featureFeed {
        features {
          ... on CommentListFeature {
            id
            comments {
              id
              person {
                id
                firstName
              }
              text
            }
          }
        }
      }
    }
  }
}	
```

Flag the comment: 
```
mutation flagComment {
  flagComment(commentId: "Comment:4f68015ba18662a7409d1219a4ce013e") {
    id
  }
}
```

Run the above `content` query again and notice the comment is no longer returned.

### What automated tests did you write?

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
